contracts/test/userAdditionalDeposit.js

```diff
@@ -0,0 +1,187 @@
+    const mintAmount = new BN('60000000000');
+    const doubleMintAmount = mintAmount.mul(new BN(2));
+    const initialAllowance = doubleMintAmount.sub(new BN(5));
+    // deposit
+    await dai.mint(accounts[1], doubleMintAmount);
+    // Here, don't mint the full 'doubleMinAmount'
+    await dai.approve(poolDeposits.address, initialAllowance, {
+      from: accounts[1],
+    });
+    let allowanceBefore = await dai.allowance.call(
+      accounts[1],
+      poolDeposits.address
+    );
+
+    const logs = await poolDeposits.deposit(mintAmount, {
+      from: accounts[1],
+    });
+    let allowanceAfter = await dai.allowance.call(
+      accounts[1],
+      poolDeposits.address
+    );
+    expectEvent(logs, 'DepositAdded', {
+      user: accounts[1],
+      amount: mintAmount,
+    });
+
+    let deposit = await poolDeposits.depositedDai.call(accounts[1]);
+
+    // check that the partial withdraw fails.
+    await expectRevert(
+      poolDeposits.deposit(mintAmount, { from: accounts[1] }),
+      'amount not available'
+    );
+
+    // User has joined the pool - no changes
+    // assert.equal(, allowance.toString());
+    assert.equal(initialAllowance.toString(), allowanceBefore.toString());
+    assert.equal(
+      initialAllowance.sub(mintAmount).toString(),
+      allowanceAfter.toString()
+    );
+    assert.equal(mintAmount, deposit.toString());
+  });
+
+  it('poolDeposits:userAdditionalDepsoit. Cannot add additional deposit more than their balance', async () => {
+    let mintAmount = new BN('60000000000');
+    let doubleMintAmount = mintAmount.mul(new BN(2));
+
+    // deposit
+    await dai.mint(accounts[1], doubleMintAmount.sub(new BN(5)));
+    await dai.approve(poolDeposits.address, doubleMintAmount, {
+      from: accounts[1],
+    });
+    let allowanceBefore = await dai.allowance.call(
+      accounts[1],
+      poolDeposits.address
+    );
+    await dai.approve(poolDeposits.address, doubleMintAmount, {
+      from: accounts[1],
+    });
+
+    const logs = await poolDeposits.deposit(mintAmount, {
+      from: accounts[1],
+    });
+    let allowanceAfter = await dai.allowance.call(
+      accounts[1],
+      poolDeposits.address
+    );
+
+    assert.equal(doubleMintAmount.toString(), allowanceBefore.toString());
+    assert.equal(mintAmount.toString(), allowanceAfter.toString());
+    expectEvent(logs, 'DepositAdded', {
+      user: accounts[1],
+      amount: mintAmount,
+    });
+
+    let deposit = await poolDeposits.depositedDai.call(accounts[1]);
+
+    // check that the partial withdraw fails.
+    await expectRevert(
+      poolDeposits.deposit(mintAmount, { from: accounts[1] }),
+      'User does not have enough DAI'
+    );
+
+    // User has joined the pool - no changes
+    assert.equal(mintAmount, deposit.toString());
+  });
+
+  // Q: do we need to implement another test checking you cannot add additional deposit once state of emergency declared?
```

Comment:
Nope that is handled already in the emergency tests I'm pretty sure :)

Reply:
I don't see tests checking dao logic based on added amounts (they can no longer vote that iteration etc...)